### ISSUE TITLE
BUG: Generic type 'ExecutionResult<TData>' requires between 0 and 1 type arguments

### ISSUE THREAD

**Issue by mgummelt** [graphile/crystal#1488]
### Summary

TypeError following https://github.com/graphile/postgraphile/commit/73fe8011d902f14c7ba29771e162bde81d6ce5b0

`Generic type 'ExecutionResult<TData>' requires between 0 and 1 type arguments`

It seems the author is assuming a type definition of `ExecutionResult` that's incompatible with graphql 14.*, but this is an accepted version of graphql in postgraphile's package.json dependencies 

### Steps to reproduce

Run postraphile 12.1.0 alongside graphql 14.7.0

**Comment by benjie** [comment]
PR welcome 👍

**Comment by davoam** [comment]
Hi, @benjie ! Thanks for the Postgraphile!  This bug breaks compiling from typescript to js, is it planned to fix it?

**Comment by benjie** [comment]
I'm focussed on PostGraphile V5; but happy to review a PR to fix it. From what I understand upgrading to GraphQL v15 is also an acceptable workaround? I've not seen this issue affect any of the projects I work on (all of which are TypeScript), so I'm not exactly sure what's involved to trigger it.

**Comment by davoam** [comment]
I solved the problem. Thanks for the quick response! 

I got the following error during the build:
```
node_modules/postgraphile/build/postgraphile/pluginHook.d.ts(80,74): error TS2707: Generic type 'ExecutionResult<TData>' requires between 0 and 1 type arguments.
```
It seems that it happened because there was this line in my `package.json` 
```
"postgraphile": "^4.5.0"
```
I ran `npm i` and `postgraphile` upgraded to version `4.12.1` where the problem appeared. 

**What helped:**
1. Fix `postgraphile` version:  `^4.5.0`  -> `4.5.5` (In my case, I was using version `4.5.5` before the problem appeared)
2. Remove `package-lock.json`
3. Run `npm i`
